packages/hop-node/src/gasboost/GasBoostTransaction.ts

```diff
@@ -693,12 +718,16 @@ class GasBoostTransaction extends EventEmitter implements providers.TransactionR
     const maxGasPrice = this.getMaxGasPrice()
     const priorityFeePerGasCap = this.getPriorityFeePerGasCap()
 
-    // don't boost if suggested gas is over max
-    const isGasPriceMaxReached = gasFeeData.gasPrice?.gt(maxGasPrice)
-    const isMaxFeePerGasReached = gasFeeData.maxFeePerGas?.gt(maxGasPrice)
-    const isMaxPriorityFeePerGasReached = gasFeeData.maxPriorityFeePerGas?.gt(priorityFeePerGasCap)
+    // don't boost if suggested gas is equal to or over max
+    const isGasPriceMaxReached = gasFeeData.gasPrice?.gte(maxGasPrice)
+    const isMaxFeePerGasReached = gasFeeData.maxFeePerGas?.gte(maxGasPrice)
```

Comment:
the updates look good; my only question/concern is here with the `gte` change.

Since this boost function calls `getBumpedGasFeeData()` and in that function [here](https://github.com/hop-protocol/hop/blob/eb7fe2c7205fe8139e7dd0159d95be812281b049/packages/hop-node/src/gasboost/GasBoostTransaction.ts#L477) it clamps it to maxGasPrice if it exceeds it, does that mean when an initial tx is sent with gas price slightly below max, then after a while it attempts to boost it;  it can't boost since it checks for `gte`  which would be true because of the clamping and it'd be marked as `isMaxReached` though the max wasn't broadcasted? 

sugggestion:
The `send` function [here](https://github.com/hop-protocol/hop/blob/eb7fe2c7205fe8139e7dd0159d95be812281b049/packages/hop-node/src/gasboost/GasBoostTransaction.ts#L373) calls `this.setGasProperties(tx)` which keeps a reference to the last used `gasPrice` and `maxFeePerGas` so maybe the boost function just needs to first check `if (this.gasPrice.gte(maxGasPrice) || this.maxFeePerGas.gte(maxGasPrice)) isMaxReached = true`? or maybe it can check if the max has been reached after it calls `_sendTransaction()`?

I could be missing something here so let me know if this doesn't make sense

Reply:
@miguelmota That is a good point, I didn't think through that. I believe what you said is true but (now that I'm re-looking at it) is expected. In that case you described, the node would reject the boosted tx for being <10% more gas than the prior tx at that nonce.

As an example with the current code (pre-PR), max gas is 500 Gwei and the first tx is 495 gwei. Gas then spikes to 1k, so the tx doesn't get included on chain. It tries to boost, but is clamped at 500 gwei. In that case, the tx will not be marked as `maxGasPriceReached` (since the clamped value `===500`), the tx will be sent again at 500 Gwei, but the node will immediately reject the tx with `transaction underpriced` and the tx will retry a few more times at 500 Gwei, eventually failing and falling out of the loop. I haven't tested that, but I think that is how it would work, if I am thinking about this correctly.

In that same example with the PR code implemented, the first tx would be attempted at 495 Gwei and then the boosted one would be marked as `maxGasPriceReached` and not attempted. The end result would be the same (no new tx making it onchain), but the process would be more efficient (wouldn't retry known-bad txs). 

Given that, I think it makes sense to keep it as-is, but happy to continue the discussion. I might have missed something here as well.

Reply:
@shanefontaine do you think there could be a scenario where it performs the initial tx (say <100 gwei) and then market gas spikes above the max (> 500gwei), so then it marks it `maxGasPriceReached` during attempted boost (and no boosted tx broadcasted) since [here](https://github.com/hop-protocol/hop/blob/9d8e7aec2fd2f3344063b490088c81917cd8df99/packages/hop-node/src/gasboost/GasBoostTransaction.ts#LL469C14-L469C35) in `getBumpedGasFeeData()` it uses the market gas price?

Reply:
@miguelmota I do believe that can happen and believe that it should be re-broadcasted at 500, which I now see it would not. 

To consolidate everything, here are the current things being tracked:

* A transfer sent at 100 Gwei should retry at 500 Gwei if the gas market spikes between the time of the initial tx and the boost
* If the initial tx is already at the max gas, that needs to be handled explicitly and not thrown into the boost logic
* `getBumpedGasFeeData()` should be consistent with clamping (or not clamping) of values
* `checkHasEnoughFunds()` should have its own error that is handled explicitly in the parent's `try...catch`. If it fails, this means the bonder is running out of funds and should be handled accordingly.

Will push a commit(s) that handles these and ping you when the updated PR is ready.